components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -18417,8 +18435,14 @@ private String addOperationPolicyContent(Connection connection, OperationPolicyD
             throws SQLException {
 
         OperationPolicySpecification policySpecification = policyData.getSpecification();
+        String toBeClonedPolicyId = policyData.getClonedCommonPolicyId();
         String dbQuery = SQLConstants.OperationPolicyConstants.ADD_OPERATION_POLICY;
-        String policyUUID = UUID.randomUUID().toString();
+        String policyUUID;
+        if (toBeClonedPolicyId == null) {
+            policyUUID = UUID.randomUUID().toString();
+        } else {
```

Comment:
You can remove this else block by directly assigning the ```policyUUID = policyData.getClonedCommonPolicyId();```

Reply:
Fixed in #[89c143a](https://github.com/wso2/carbon-apimgt/pull/11495/commits/89c143aa9c820c3542909928a10a8189ffee0d02)